### DISCUSSION TITLE
You can't just decide stuff and check it in

### DISCUSSION THREAD

**Discussion by DominikPinsel** [eclipse-tractusx/ssi-docu#28] [General]
Currently, this repository has many specific technical details, like protocols and SSI components, that the MIW hasn't agreed on yet.

Documentation for SSI is clearly needed, and I agree that having a dedicated feature for exchanging Verifiable Credentials, as explained in this repository, is important. But this feature matters not only to the EDC team (who wrote most of this documentation), but also to other stakeholders. The MIW team needs to evaluate and decide the best solution for integrating it into the MIW.

Here are a few examples that emphasize these points:

- This repository introduces something called the 'Credential Service.' The way it works shows it's different from the MIW. I think it might be the Identity Hub. However, we're still deciding whether to include an identity hub in the Catena-X ecosystem, so it's not confirmed. We need to consider various things, like security, resources, how it benefits all stakeholders, and more.
- The choice of protocol is another issue. While the MIW is still discussing which protocol to use, this repository has already chosen the [Self-Issued OpenID Provider v2 specification](https://openid.net/specs/openid-connect-self-issued-v2-1_0.html#section-11.1).
- There's a concern about the security of the Credential Service endpoint, involving an "[OAuth 2](https://www.rfc-editor.org/rfc/rfc6749) access token obtained from the resource owner." If this is related to an Identity Hub, it's better to follow the official specification. If it's a custom service without an official spec, we should stick to the principle of using recognized standards. Going against this could lock us into one vendor, which we want to avoid.

I want to clarify that my concerns aren't about everything in this repository, just specific parts. Many topics haven't been discussed or agreed upon, and some haven't even involved SSI experts. So, I won't review those areas, as that might give them undue credibility. This isn't how things usually work.

To address this, I suggest one of these options:

- Transfer the majority of the repository's content to a distinct branch. This content can then be subject to rigorous discussion before being merged into the main branch.
- Alternatively, consider deleting this repository. It's not official and it is causing confusion. The things document here are ether already implemented, or not agreed upon. We don't need or want it.

Lastly, make sure to clearly state in a disclaimer that this repository isn't official documentation. Accuracy is crucial, and this distinction needs to be evident.

**Comment by jimmarino** [comment]
Sorry, but for some reason, you are not aware of the purpose of this repository. The decision was made by the stakeholders (including those responsible for the MiW) that this repository would be the home of the official specifications for Tractus-X SSI. In addition, the decision was made to start the SSI protocols on the approach successfully adopted by the EDC.

The SSI specifications are based on existing standards where appropriate. There have been numerous discussions that have explained this in detail. The specifications will be evolved, but projects, including EDC and MiW, will implement these specifications.

If you have specific technical concerns, please either open a detailed issue for discussion and/or work with the MiW representative to the spec group.
